TESTING.md

```diff
@@ -0,0 +1,115 @@
+
+Here is an example test that checks if labels are present in the statefulset.
+
+https://github.com/prometheus-operator/prometheus-operator/blob/aeceb0b4fadc8307a44dc55afdceca0bea50bbb0/pkg/prometheus/server/statefulset_test.go#L140-L167
+
+Unit tests can be run with:
+
+```
+make test-unit
+```
+
+They can also be run for particular packages:
+
+```
+go test ./pkg/prometheus/server
+```
+
+Or even particular functions:
+
+```
+go test -run ^TestPodLabelsAnnotations$ ./pkg/prometheus/server
+```
+
+### Testing multiline string comparison - Golden files
+
+Golden files are plaintext documents designed to facilitate the validation of lengthy strings. They come in handy when, for instance, you need to test a Prometheus configuration that's generated using Go structures. You can marshal this configuration into YAML and then compare it against a static reference to ensure a match. Golden files offer an elegant solution to this challenge, sparing you the need to hard-code the static configuration directly into your test code.
+
+In the example below, we're generating the Prometheus configuration (which can easily have 100+ lines for each individual test) and comparing it against a golden file:
+
+https://github.com/prometheus-operator/prometheus-operator/blob/aeceb0b4fadc8307a44dc55afdceca0bea50bbb0/pkg/prometheus/promcfg_test.go#L102-L277
+
+If not for golden files, the test above, instead of ~150 lines, would easily require around ~1000 lines. The usage of golden files help us maintain test suites with several multiline strings comparison without sacrifing test readability.
+
+### Updating Golden Files
+
+There are contributions, e.g. adding a new required field to an existing configuration, that requires us to update several golden files at once. This can easily be done with the command below:
+
+```
+make test-unit-update-golden
+```
+
+## End-to-end tests
+
+Sometimes, running tests in isolation is not enough and we really want test the behavior of Prometheus-Operator when running in a working Kubernetes cluster. For those occasions, end-to-end tests are our choice.
+
+To run e2e-tests, first start a Kubernetes cluter. We recommend using [MiniKube](https://minikube.sigs.k8s.io/docs/start/) or [KinD](https://kind.sigs.k8s.io/) for development purposes since they are light weight distributions of kubernetes and can run on small notebooks.
+
+For manual e2e tests, you can use the utility script [scripts/run-external.sh](./scripts/run-external.sh), it will make sure all the requirements are in place to run Prometheus-Operator without problems, but for automated end-to-end tests, we have the command:
```

Comment:
```suggestion
For manual testing, you can use the utility script [scripts/run-external.sh](./scripts/run-external.sh), it will check all the requirements and run your local version of the Prometheus Operator on your Kind cluster:

`
./scripts/run-external.sh -c
`

For automated end-to-end tests, we have the command:
```